src/extensions/mvs/components/primitives.ts

```diff
@@ -582,10 +665,103 @@ function addLineMesh(context: PrimitiveBuilderContext, { groups, mesh }: MeshBui
         const count = Math.ceil(dist / (2 * params.dash_length));
         addFixedCountDashedCylinder(mesh, lStart, lEnd, 1.0, count, true, cylinderProps);
     } else {
+        // NOTE: this: use similar
         addSimpleCylinder(mesh, lStart, lEnd, cylinderProps);
     }
 }
 
+function addBoxMesh(context: PrimitiveBuilderContext, { groups, mesh }: MeshBuilderState, node: MVSNode<'primitive'>, params: MVSPrimitiveParams<'box'>, options?: { skipResolvePosition?: boolean }) {
+    // TODO: need this?
+    // if (!options?.skipResolvePosition) {
+    //     // TODO: need this?
+    //     resolveBasePosition(context, params.start, lStart);
+    //     resolveBasePosition(context, params.end, lEnd);
+    // }
+
+    const a = Vec3(), b = Vec3(), c = Vec3(), d = Vec3();
+    // TODO: as_edges, rotation, edge_radius => cage, rotation matrices
+    const { center, extent, box_groups, scaling, as_edges, edge_radius, rotation, translation } = params;
```

Comment:
You should call `resolvePosition` on the `center` to get a bounding box to which you add the extent and possibly multiple by the scaling.

Reply:
@dsehnal, thanks. Do you mean `resolveBasePosition`?

Reply:
As a use case (pseudocode), this should work:

```
structure = builder()
  .download("1tqn")
  .model()
  .structure()

structure.component("ligand").representation("ball-and-stick")
structure.component("polymer").representation("cartoon")

structrure
  .primitives()
  .box(center={ auth_seq_id: 508 }, extent=[2,2,2])  # or .cage
```

This should draw a box around the HEM residue + 2ang in each direction.